src/System.Xml.XmlSerializer/src/System/Xml/Serialization/XmlReflectionImporter.cs

```diff
@@ -2006,7 +2014,7 @@ private void CheckAmbiguousChoice(XmlAttributes a, Type accessorType, string acc
                     }
                     else
                     {
-                        choiceTypes.Add(type, false);
+                        choiceTypes.Add(type);
```

Comment:
Nit: Same comment about using single call to `HashSet<T>.Add` instead of `Contains` & `Add`.